Added transform listener and its methods

HiroIshida opened the pull request:
I added transform listener and those methods: `lookupTransform` and `waitForTransform`.

### style 
To implement `TransformListener`, I followed the style in [this](https://github.com/jdlangs/RobotOS.jl/blob/73e79aae5a2c2e3ba5ba9b8e3fea0fc78a0cb915/src/pubsub.jl#L11). For implementation of `waitForTransform` and `lookupTransform`, I follow the style in [this](https://github.com/jdlangs/RobotOS.jl/blob/73e79aae5a2c2e3ba5ba9b8e3fea0fc78a0cb915/src/services.jl#L58). 

### test
To test this new feature, script `test/tfpubnode.py` runs and keep publishing a specified tf `([0, 0, 0], [0, 0, 0, 1])`. In the meantime, `test/tf.jl` runs and `waitforTransform` and `lookupTransform` is called and tested if the specified tf is actually obtained. 

### travis
Because to use tf we need `ros-kinetic-tf2-ros` and `ros-kinetic-tf` packages, I added installation of these packages inside `.travis.yml`. 

### suggestion on exception handling  
Like in the [tutorial of tf](http://wiki.ros.org/tf/Tutorials/Writing%20a%20tf%20listener%20%28Python%29), It is fairly common to use tf listener with some exception handling. To do this, we need to check the type of exception. Therefore, it is convenient to have exception types. For example, how about [this implementaion](https://github.com/HiroIshida/RobotOS.jl/commit/e74979efa9f5146a3760bcce1bd3c1e7430b6fde#diff-8c1e4a260a6813e95454f81648395273
)(my another branch) for throwing `LookupException`? Aside from the convenience, having these exception types is important to do exception handling test. For example, to test `waiforTransform` works correctly, I think we have to test the case where `LookupTranform` throws `LookupException` without `waitForTransform`.  I would like to hear your opinion on this.

HiroIshida committed:
Merge remote-tracking branch 'upstream/master'

HiroIshida committed:
add TransformListener ans its method

HiroIshida committed:
add waitForTransform

HiroIshida committed:
add doc

HiroIshida committed:
catch exception

HiroIshida committed:
show error massage include exception type information

HiroIshida committed:
added test and check passed

HiroIshida committed:
fix typos

HiroIshida committed:
add apt install ros-kinetic-tf2-ros and ros-kinetic-tf in travis

jdlangs commented:
Hi @HiroIshida, I appreciate your work on this added functionality. In the past, I've been reluctant to start adding functions from ROS packages other than `rospy` just since the line starts to get really blurry about what does and doesn't belong here. That said, I suppose `tf` is pretty unique and if anything deserves to get a Julia API, it's probably `tf`.

A few thoughts to keep moving forward here:
- I would prefer anything `tf`-related get put into a a `TF` submodule underneath `RobotOS` so it's quite clear this is separate functionality.
- It would be nice to also provide transform broadcasting too but if you don't have the bandwidth to add that I won't hold up the PR.
- For exceptions, I'm fine with creating and throwing a TF exception type in the Julia code
- I would strongly prefer not to provide transform data as an ad-hoc tuple/array return value, but as a proper Julia type. Since we can't easily use `geometry_msgs` types here, I think it would be fine to define a transformation type yourself.

Let me know if you have further thoughts.

HiroIshida commented:
Hi @jdlangs, thank you very much for your comment. I understand the whole points. Yes, `tf.broadcastor` and related things are also important functionarities and I actually planned to add those. So I will do it! I will be bit busy until 5/10. But after that, I will do that in few days.

HiroIshida committed:
tranform as a submodule

HiroIshida committed:
add Transform type

HiroIshida committed:
add test for Transform type

HiroIshida committed:
add tf broadcaster

HiroIshida committed:
add test for tf broadcaster

HiroIshida committed:
update travis

HiroIshida committed:
fix index in response to the change of test sequence

HiroIshida committed:
add comments for tf broadcaster

HiroIshida committed:
Merge branch 'comment_reponse' into add_tf

HiroIshida committed:
chmod +x

HiroIshida commented:
### Changes
1) First I made a submodule named `TF`.
2) add `TransformBroadcaster` and `sendTransform`
3) add `Transform` type. 
4) fix test sequence in response to change 2. Previously, I executed a node in `tfpubnode.py`, and the test was conducted by checking if the received transform is the same as the published one. But I deleted this node, and instead of that, I run `tfbroadcastnode.jl` which broadcaster the tf by using `sendTransform`. 
5) in response to change 4, I updated the `index.md` so that the user will execute `test/tfbroadcastnode.jl` as well as `test/echonode.py` before running `runtests.jl`.  

### Concerns and questions
a) as for change 3, I'm not sure my implementation of `Transform` is what you meant. 
b) I mentioned about adding custom error type, and I made several attempt on that, but I had difficulty in testing these exceptions. And the test code for that must be complex. So, this time I stop adding exception types. 
c) The test for my update on travis failed. I guess `julia test/tfbroadcastnode.jl` is properly executed. But I don't know why this happens. Could anybody help me?
